### PR TITLE
fix(ingest/bq): fix ordering of queries for use_queries_v2

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py [hunk 1]
```diff
@@ -249,14 +249,28 @@ def get_workunits_internal(
...
+            )
 
         with self.report.audit_log_load_timer:
             i = 0
-            for query_instances in queries_deduped.values():
-                for _, query in query_instances.items():
+            for queries_in_bucket in queries_deduped.values():
+                # Assuming that FileBackedDict is an OrderedDict.
+                # If this is not the case, we would need to order it by "last_query_timestamp"
+                # using queries_in_bucket.sql_query_iterator.
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py:272]: i'm not sure this assumption is valid - might need to write some tests to valid this, but I think sqlite doesn't guarantee anything

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py:272]: yeah we will need to tweak the code here https://github.com/datahub-project/datahub/blob/3d5735cbc5899bf294ab3858a78cb9094dea3136/metadata-ingestion/src/datahub/utilities/file_backed_collections.py#L353

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py]: I'm not superfamilar with file_backed_collections code. I updated the queries_extractor to use order by sql query when reading the results.

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py [hunk 2]
```diff
@@ -267,10 +267,11 @@ def get_workunits_internal(
...
             for queries_in_bucket in queries_deduped.values():
-                # Assuming that FileBackedDict is an OrderedDict.
-                # If this is not the case, we would need to order it by "last_query_timestamp"
-                # using queries_in_bucket.sql_query_iterator.
-                for query in queries_in_bucket.values():
+                # Ordering is essential for column-level lineage via temporary table
+                for row in queries_in_bucket.sql_query_iterator(
+                    "select value from data order by last_query_timestamp asc",
+                ):
+                    query = queries_in_bucket.deserializer(row["value"])
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/queries_extractor.py:274]: this is fine, but it'd be even better if we just modified `FileBackedDict` to handle this properly

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
@mayurinehate doing this improvement here https://github.com/datahub-project/datahub/pull/11346

**Pull request merged**